test/inductor/test_unbacked_symints.py

```diff
@@ -294,6 +294,95 @@ def fn(value, mask):
         expected = fn(*example_inputs)
         torch.testing.assert_close(actual, expected)
 
+    @dynamo_config.patch({"capture_scalar_outputs": True})
+    @parametrize("dynamic", [False, True, None])
+    def test_unbacked_slice_on_subclass(self, device, dynamic):
+        from torch.testing._internal.common_subclass import WrapperTensor
+        from torch.utils._pytree import tree_map
+
+        # NB: the error we're testing for only triggers when unbacked SymInts
+        # are created within a subclass's torch_dispatch, because they're not seen
+        # by Dynamo and thus are considered freshly-created when the subclass instance
+        # return value of the torch_dispatch is handled.
+        # Subclass forwards everything along to the single underlying dense tensor
+        # component, except for slice(), which it handles via data-dependent bounds access
+        class CustomSliceSubclass(WrapperTensor):
+            @classmethod
+            def get_wrapper_properties(cls, t, slice_bounds=None):
+                return t, {}
+
+            def __init__(self, t, slice_bounds=None):
+                self.t = t
+                self.slice_bounds = slice_bounds
+
+            def __repr__(self):
+                t_repr = repr(self.t)
+                slice_bounds_repr = repr(self.slice_bounds)
+                return f"CustomSliceSubclass({t_repr}, {slice_bounds_repr})"
+
+            def __tensor_flatten__(self):
+                return ["t", "slice_bounds"], None
+
+            @classmethod
+            def __tensor_unflatten__(
+                cls, inner_tensors, meta, outer_size, outer_stride
+            ):
+                t = inner_tensors["t"]
+                slice_bounds = inner_tensors["slice_bounds"]
+                return cls(t, slice_bounds)
+
+            @classmethod
+            def __torch_dispatch__(cls, func, types, args=(), kwargs=None):
+                if func is torch.ops.aten.slice.Tensor:
```

Comment:
So, if I understand correctly, when we do Dynamo tracing, we will end up with a torch.ops.aten.slice.Tensor node in the graph, which is taking the subclass in as an argument. So for the purpose of `unbacked_binding` calculation, we must be able to determine the allocated unbacked SymInts from the returned tensor of this subclass slice operation?

Reply:
> we will end up with a torch.ops.aten.slice.Tensor node in the graph, which is taking the subclass in as an argument.

yep, this is correct.

> So for the purpose of unbacked_binding calculation, we must be able to determine the allocated unbacked SymInts from the returned tensor of this subclass slice operation?

yep that's my understanding as well. `compute_unbacked_bindings()` checks size / stride / storage_offset of subclass outer inner tensors for these unbacked SymInts. In particular for this subclass, `t.storage_offset()` will be `<something> * u1` where `<something>` is a constant when `dynamic=False` and a backed SymInt when `dynamic=True`.

Reply:
My point is that you shouldn't be detecting the offset from storage_offset. IMO, it should just be stored directly somewhere.